core/src/SYCL/Kokkos_SYCL_Instance.hpp

```diff
@@ -70,18 +70,157 @@ class SYCLInternal {
 
   std::unique_ptr<cl::sycl::queue> m_queue;
 
+  // USMObjectMem is a reusable buffer for a single object
+  // in USM memory
+  template <sycl::usm::alloc Kind>
+  class USMObjectMem {
+   public:
+    static constexpr sycl::usm::alloc kind = Kind;
+
+    friend void swap(USMObjectMem& lhs, USMObjectMem& rhs) noexcept {
+      using std::swap;
+      swap(lhs.m_q, rhs.m_q);
+      swap(lhs.m_data, rhs.m_data);
+      swap(lhs.m_capacity, rhs.m_capacity);
+    }
```

Comment:
It'll be slightly more efficient, as `sycl::queue` is a `shared_ptr` under the covers, although it is there because I just tend to add a hidden friend swap function when writing value types more than having given it much thought.